src/components/FlagsProvider.tsx

```diff
@@ -26,7 +26,9 @@ const FlagsProvider = ({children, config}: FlagsProviderProps) => {
         // @deprecated use host instead
         url: process.env.REACT_APP_FLAGS_CTX_URL || '',
         // uri to the unleash API '/client/features
-        uri: process.env.REACT_APP_FLAGS_CTX_URI || DEFAULT_FEATURES_URI
+        uri: process.env.REACT_APP_FLAGS_CTX_URI || DEFAULT_FEATURES_URI,
+        // the userId to check strategies against
+        userId: process.env.REACT_APP_FLAGS_CTX_USER_ID || '' 
```

Comment:
Chances of knowing the user ID at this point are really small. As suggested, this should be handled in a context which is passed to the client at the moment we need to check a flag.